nodes/viz/viewer_blf_text.py

```diff
@@ -0,0 +1,370 @@
+        - showing background polygon may be moderately more processor intensive, as it must first calculate all polygon sizes
+          and locations and draw them all first, then draw text over all of them.
+
+    """
+
+    bl_idname = 'SvViewerTextBLF'
+    bl_label = 'Viewer blf Text'
+    bl_icon = 'INFO'
+    sv_icon = 'SV_INDEX_VIEWER'
+
+    def make_color_prop(name, col):
+        return FloatVectorProperty(
+            name=name, description='', size=4, min=0.0, max=1.0,
+            default=col, subtype='COLOR', update=updateNode)
+
+    activate: BoolProperty(
+        name='Show', description='Activate node?',
+        default=True, update=updateNode)
+
+    draw_background: BoolProperty(
+        name='draw_background', description='draw background polygons or not',
+        default=False, update=updateNode)
+
+    coordinate_rounding: IntProperty(
+        name="rounding", default=3, min=0, soft_max=8, 
+        description="in coordinate mode, Use this slider to adjust how precise you want to display each coordinate",
+        update=updateNode)
+
+    skip_empty_strings: BoolProperty(name="Skip empty Strings", update=updateNode)
+
+    anchor_direction: EnumProperty(
+        items=enum_item_5(
+            ["left", "right", "center", "top", "bottom"], 
+            ["ANCHOR_LEFT", "ANCHOR_RIGHT", "ANCHOR_CENTER", "ANCHOR_TOP", "ANCHOR_BOTTOM"]),
+        description="offers a way to anchor all text items to the location",
+        default="center", update=updateNode)
+
+    background_color: make_color_prop("background_color", (.2, .2, .2, 1.0))
+    font_text_color: make_color_prop("font_text_color", (0.6, 1, 0.3, 1.0))
+    text_node_scale: FloatProperty(name="text scale", default=1.0, min=0.1, update=updateNode)
+
+    def sv_init(self, context):
+        inew = self.inputs.new
+        inew('SvVerticesSocket', 'locations')
+        inew('SvStringsSocket', 'text')
+
+
+    def draw_buttons(self, context, layout):
+        view_icon = 'RESTRICT_VIEW_' + ('OFF' if self.activate else 'ON')
+
+        column_all = layout.column()
+
+        row = column_all.row(align=True)
+        split = row.split()
+        r = split.column()
+        r.prop(self, "activate", text="Show", toggle=True, icon=view_icon)
+        row.prop(self, "draw_background", text="BG", toggle=True)
+        new_row = layout.row()
+        new_row.prop(self, "anchor_direction", expand=True)
+
+
+    def get_settings_dict(self):
+        
+        """
+        Produce a dict of settings for the callback
+            
+        A copy is needed, we can't have reference to the node in a callback, 
+        it will crash blender on undo
+        """
+
+        return {
+            'background_color': self.background_color[:],
+            'font_text_color': self.font_text_color[:],
+            'draw_background': self.draw_background,
+            'anchor': self.anchor_direction,
+            'scale': self.text_node_scale
+        }.copy()
```

Comment:
Did not catch sense of `copy` here

Reply:
we established early on that passing references from a node into a callback made things unstable, sporadically. i'm not prepared to drop the copy. worry about bigger things.

Reply:
Sound like magic. Magic is not good in coding.

Reply:
if it isn't needed it can of course go. the docstring explains why it's there (or why it was initially added to such callbacks). it may not be needed anymore, but you might want to dig through the issue tracker to see when it was first added and why and by who.

not the focus of this PR.

Reply:
You probably know that copy only create copy of a dictionary and not its contents. Your dictionary is initialized in the method. So every time the method will be called the new dictionary will be returned. 

I know if a programmer treats code as magic it leads to unpredictable results.
https://docs.python.org/3/library/stdtypes.html#dict.copy

Reply:
thank you for your insight.

Reply:
i'll remove `copy`, and also other similar instances in other viewers. This hasn't been discussed since 2014

https://github.com/nortikin/sverchok/issues/401#issuecomment-55401403